docs/Model Pipeline.md

```diff
@@ -0,0 +1,364 @@
+
+The nature of the modification is flexible and defined by the model. Some model plugins will calculate an impact metric (for example, energy from utilization) and add that to the observation before passing that to the next model. Some models will enrich the observation with meta-data required for other models, for example, adding grid carbon intensity values.
+
+## Initialization
+
+All the model plugins used in any component in the graph are configured at the top of the [Impl (Impact YAML)](Impl%20(Impact%20YAML).md) in the initialize section. There are multiple places to add configuration for model plugins; any configuration added here is global and applied to every use of the model throughout the graph.
+
+```yaml
+.
+.
+config:
+  models:
+    - name: <model-name-1>
+      kind: builtin | plugin | shell
+      path: <path-to-ts-module-to-load>
+      config:
+	       <key>: <value>
+    - name: <model-name-2>
+      kind: builtin | plugin | shell    
+      path: <path-to-ts-module-to-load>
+      config:
+	       <key>: <value>	       
+.
+.
+graph: ~
+```
+
+## Usage
+
+In component nodes, we then configure the models we want to use in a pipeline like so:
+
+```yaml
+backend:
+  pipeline:
+    - model-1
+    - model-2
+    - model-3  
+  config: 
+    model-1:
+      <key>: <value>
+    model-2:
+      <key>: <value>        
+    model-3:
+      <key>: <value>                
+  observations: 
+    - timestamp: 2023-07-06T00:00
+      duration: 5
+      cpu: 33
+    - timestamp: 2023-07-06T00:05
+      duration: 5
+      cpu: 23
+    - timestamp: 2023-07-06T00:10
+      duration: 5
+      cpu: 11
+```
+
+
+- `pipeline` defines the models we apply and the order in which we use them.
+- `config` in this part of the graph is config for each model in the pipeline (if any is required). Since we have multiple models, we need to define each config independently.
+- `observations` are the source observations, the values we are pumping into the start of this pipeline.
+
+
+## Example
+
+Let's look at a simple pseudo example, to begin with a pipeline like so:
+
+```yaml
+pipeline:
+  - instance-metadata
+  - tdp
+  - teads-curve
+```
+
+Using the above, we can combine multiple smaller models together to calculate the energy consumed by this observation:
+
+```yaml
+observations: 
+  - timestamp: 2023-07-06T00:00
+    vendor: aws
+    instance-type: m5d.large
+    duration: 5
+    cpu-util: 33
+```
+
+### `instance-metadata`
+
+This model plugin takes as input an *instance type* and outputs the name of the physical processor of the underlying architecture as well as other valuable metadata, like so:
+
+```yaml
+observations: 
+  - timestamp: 2023-07-06T00:00
+    vendor: aws
+    instance-type: m5d.large
+    duration: 5
+    cpu-util: 33
+```
+
+to 
+
+```yaml
+observations: 
+  - timestamp: 2023-07-06T00:00
+    vendor: aws
+    instance-type: m5d.large
+    duration: 5
+    cpu-util: 33

```

Comment:
Having hyphens in middle causes the parsers to behave differently and especially in javascript, the object access syntax we use will be void and we need to use the array syntax.